src/Func.cpp

```diff
@@ -888,7 +889,9 @@ Func Func::in(const Func &f) {
             << "Redefinition of shared wrapper with " << it.first << " [" << name() << " -> "
             << Function(wfunc).name() << "] in " << f.name() << " is not allowed\n";
     }
-    return Func(Function(wfunc));
+    Func wrapper = Func(Function(wfunc));
```

Comment:
This would be slightly simpler:

    Function wrapper(wfunc);
    internal_assert(wrapper.frozen());
    return Func(wrapper);